src/spectral.ts

```diff
@@ -135,6 +137,17 @@ export class Spectral {
     }
   }
 
+  private setExceptions(exceptions: RulesetExceptionCollection) {
+    const target: RulesetExceptionCollection = {};
+    mergeExceptions(target, exceptions);
+
+    empty(this.exceptions);
+
+    Object.entries(target).forEach(([location, rules]) => {
```

Comment:
```Object.assign(this.exception, target)``` should be sufficient I believe? It doesn't create a new array for rules though, but if you do need it, you can just clone the target beforehand.

Reply:
@P0lip Fixed